test/DynamoCoreTests/NodeExecutionTest.cs

```diff
@@ -0,0 +1,394 @@
+        {
+            var model = CurrentDynamoModel;
+            //create some numbers
+            var numberNode1 = new DoubleInput();
+            numberNode1.Value = "1";
+            var numberNode2 = new DoubleInput();
+            numberNode2.Value = "2";
+            var addNode = new DSFunction(model.LibraryServices.GetFunctionDescriptor("+"));
+
+            model.ExecuteCommand(new DynamoModel.CreateNodeCommand(numberNode1, 0, 0, true, false));
+            model.ExecuteCommand(new DynamoModel.CreateNodeCommand(numberNode2, 0, 0, true, false));
+            model.ExecuteCommand(new DynamoModel.CreateNodeCommand(addNode, 0, 0, true, false));
+
+            //connect them up
+            model.ExecuteCommand(new DynamoModel.MakeConnectionCommand(numberNode1.GUID, 0, PortType.Output, DynCmd.MakeConnectionCommand.Mode.Begin));
+            model.ExecuteCommand(new DynamoModel.MakeConnectionCommand(addNode.GUID, 0, PortType.Input, DynCmd.MakeConnectionCommand.Mode.End));
+
+            model.ExecuteCommand(new DynamoModel.MakeConnectionCommand(numberNode2.GUID, 0, PortType.Output, DynCmd.MakeConnectionCommand.Mode.Begin));
+            model.ExecuteCommand(new DynamoModel.MakeConnectionCommand(addNode.GUID, 1, PortType.Input, DynCmd.MakeConnectionCommand.Mode.End));
+
+            Assert.AreEqual(model.CurrentWorkspace.Nodes.Count(), 3);
+            Assert.AreEqual(model.CurrentWorkspace.Connectors.Count(), 2);
+
+            //check the value
+            AssertPreviewValue(addNode.GUID.ToString(), 3);
+
+            //freeze add node
+            addNode.IsFrozen = true;
+           
+            // the add node must be frozen and not executing state
+            Assert.AreEqual(addNode.IsFrozen, true);
+           
+            //freeze number node.
+            numberNode1.IsFrozen = true;
+            
+            // the number node must be frozen and not executing state
+            Assert.AreEqual(numberNode1.IsFrozen, true);
+            
+            //change the value on number node 1
+            numberNode1.Value = "3.0";
+
+            // the add node must be frozen and in a temporary state
+            Assert.AreEqual(addNode.IsFrozen, true);
+           
+            //check the value on add node.
+            AssertPreviewValue(addNode.GUID.ToString(), 3);
+
+            //now unfreeze number node.
+            numberNode1.IsFrozen = false;
+           
+            //now number node is not frozen
+            Assert.AreEqual(numberNode1.IsFrozen, false);
+
+            //this causes the add node to switch back from temporary state
+            //to frozen and not executing state
+            Assert.AreEqual(addNode.IsFrozen, true);
+            
+            //check the value on add node. 
+            AssertPreviewValue(addNode.GUID.ToString(), 5);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void NodeAttachedTThatHasParentsShouldNotUnfreezeUntilAllParentsUnfreeze()
```

Comment:
Check spelling/duplicate letters in all these test names. If it helps, make the test names "_" separated like `NodeAttached_HasParents_ShouldNot_UnfreezeUntilParentsUnfreeze`